apps/src/javalab/JavalabConsole.jsx

```diff
@@ -61,11 +63,44 @@ class JavalabConsole extends React.Component {
     commandHistory: new CommandHistory(),
   };
 
+  hasCompilationError = log => {
+    return log.text && log.text.includes(javalabMsg.compilerError());
+  };
+
+  hasCompilationSucces = log => {
+    return log.text && log.text.includes(javalabMsg.compilationSuccess());
+  };
+
+  checkForCompilationError = consoleLogs => {
+    const {setHasCompilationError} = this.props;
+    const logsSortedByRecency = consoleLogs.slice().reverse();
+    const indexOfMostRecentError = logsSortedByRecency.findIndex(
+      this.hasCompilationError
+    );
+    if (indexOfMostRecentError !== -1) {
+      const logsAfterError = logsSortedByRecency.slice(
+        0,
+        indexOfMostRecentError
+      );
+      const hasUnresolvedError = !logsAfterError.find(
+        this.hasCompilationSucces
+      );
+      setHasCompilationError(hasUnresolvedError);
```

Comment:
You could also search the logs from the back and see whether you hit a compilation success or a compilation error first, I think, right? I think that would save you from having to sort and compare the two indices, simplifying the code a little, but it doesn't matter to me whether you change it.